src/bdc/steps/google_places.py

```diff
@@ -34,50 +69,212 @@ def verify(self) -> bool:
+                search_query = re.sub(r"[^a-zA-Z0-9\n]", " ", account_name)
+
+        if search_query is None and phone_number is None:
+            # if account name consists only of first and last name and no custom domain is available,
+            # skip the search as no results are expected
+            return error_return_value
+
+        response_by_mail, response_count_mail = self.get_first_place_candidate(
+            search_query, "textquery"
+        )
+
+        response_by_phone, response_count_phone = self.get_first_place_candidate(
+            phone_number, "phonenumber"
+        )
+
+        # compare the place_id, if it doesn't match just output results by email for now
+        if response_by_mail is not None and response_by_phone is not None:
+            place_id_matches_phone_search = (
+                response_by_phone["place_id"] == response_by_mail["place_id"]
+            )
+        else:
+            place_id_matches_phone_search = False
+
+        chosen_response = (
+            response_by_mail if response_by_mail is not None else response_by_phone
+        )
+
+        if chosen_response is None:
+            return error_return_value
 
-        # Go through each email address entry and remove the domain name (can do this in preprocessing, this is for test)
-        domain = lead_row["domain"]
-        if domain is None:
+        results_list = [
+            chosen_response[field] if field in chosen_response else None
+            for field in self.api_fields
+        ]
+
+        # add number of candidates, which is not a direct field in the api response but can be derived from it
+        results_list.append(response_count_mail)
+        results_list.append(response_count_phone)
+
+        # add boolean indicator whether search by phone result matches search by email
+        results_list.append(place_id_matches_phone_search)
+
+        # calculate confidence score for google places results
+        results_list.append(self.calculate_confidence(results_list, lead_row))
+
+        return pd.Series(results_list)
+
+    def get_data_from_detailed_google_api(self, lead_row):
+        error_return_value = pd.Series([None] * len(self.detailed_df_fields))
+
+        place_id = lead_row["google_places_place_id"]
+
+        if place_id is None or pd.isna(place_id):
             return error_return_value
 
+        # Call for the detailed API using specified fields
         try:
-            response = self.gmaps.find_place(domain, "textquery", fields=self.fields)
+            # Fetch place details including reviews
+            response = self.gmaps.place(place_id, fields=self.detailed_api_fields)
+
+            # Check response status
+            if response.get("status") != HTTPStatus.OK.name:
+                self.log(f"Failed to fetch data. Status code: {response.get('status')}")
+                return error_return_value
+
+        except RequestException as e:
+            self.log(f"Error: {str(e)}")
```

Comment:
Maybe merge felixs' branch into yours first, as he has implemented a logger module already

Reply:
Yep, he merged before me, so I just handled it in the merge conflicts